wagtail/admin/compare.py

```diff
@@ -55,8 +57,155 @@ def htmldiff(self):
+                comparisons.append(comparison_class(block.block, a_blocks_by_id[block.id].value, block.value))
+            else:
+                # New block
+                comparisons.append(comparison_class(block.block, None, block.value))
+
+        # Insert deleted blocks at the index where they used to be
+        deleted_block_indices = [(block, i) for i, block in enumerate(a_blocks) if block.id in deleted_ids]
+
+        for block, index in deleted_block_indices:
+            comparison_class = get_comparison_class_for_block(block.block)
+
+            # Insert the block back in where it was before it was deleted.
+            # Note: we need to account for new blocks when finding the position.
+            current_index = 0
+            block_inserted = False
+            for i, comparison in enumerate(comparisons):
+                if comparison.is_new():
+                    continue
+
+                if current_index == index:
+                    comparisons.insert(i, comparison_class(block.block, block.value, None))
+                    block_inserted = True
+                    break
+
+                current_index += 1
+
+            # Deleted block was from the end
+            if not block_inserted:
+                comparisons.append(comparison_class(block.block, block.value, None))
+
+        return comparisons
+
+    def htmldiff(self):
+        comparisons_html = []
+
+        for comparison in self.get_block_comparisons():
+            classes = ['comparison__child-object']
+            if comparison.is_new():
+                classes.append('addition')
+                block_rendered = comparison.block.render_basic(comparison.val_b)
+            elif comparison.is_deleted():
+                classes.append('deletion')
+                block_rendered = comparison.block.render_basic(comparison.val_a)
+            elif comparison.has_changed():
+                block_rendered = comparison.htmldiff()
+            else:
+                block_rendered = comparison.block.render_basic(comparison.val_a)
+
+            classes = ' '.join(classes)
+            comparisons_html.append(f'<div class="{classes}">{block_rendered}</div>')
+
+        return mark_safe(''.join(comparisons_html))
+
+
+class StreamFieldComparison(FieldComparison):
+    def has_block_ids(self, val):
+        if not val:
+            return True
+
+        return bool(val[0].id)
+
+    def htmldiff(self):
+        # Our method for diffing streamfields relies on the blocks in both revisions having UUIDs.
+        # But as UUIDs were added in Wagtail 1.11 we can't compare revisions that were created before
+        # that Wagtail version.
+        if self.has_block_ids(self.val_a) and self.has_block_ids(self.val_b):
+            return StreamBlockComparison(self.field.stream_block, self.val_a, self.val_b).htmldiff()
+        else:
+            # Fall back to diffing the HTML representation
+            return diff_text(
```

Comment:
I'm about to make some improvements to RichTextBlockComparison but I'd still like the fallback for streamfields to keep using this simpler method since those improvements won't be useful for non-rich-text fields.